### PR TITLE
[readable-stream] Fix working on TS < 5.2

### PR DESCRIPTION
**Pull request by Semigradsky**
Closes #67079

### CODE REVIEW

**Comment by jakebailey** [comment]
Hmm, I'm not sure that a npm package should be declaring the existence of global symbols like this... But I'm not totally sure what a library can do today to conditionally declare this for old libs either... @rbuckton

**Comment by Semigradsky** [comment]
> There are ways to define the method conditionally, though it's not pretty:

There is a compile error in typescript@4.6

**Comment by jakebailey** [comment]
We can raise the minimum version of the package if it's a problem. 4.6 is nearly unsupported anyway.

I don't think we should be changing the dependency version to something other than `*` (in the latest update).

**Comment by jakebailey** [comment]
Hm, yeah, I guess quite a few packages depend on this one. They'd all have to be updated. Perhaps that's not the right fix either... (It's the weekend, I'll ask around tomorrow)

File: types/readable-stream/index.d.ts [hunk 1]
```diff
@@ -8,6 +8,11 @@ declare class StringDecoder {
     end(buffer?: Buffer): string;
 }
 
+declare var NoAsyncDispose: {
+    new(...arguments: any[]): typeof globalThis.Symbol extends { readonly asyncDispose: infer S extends symbol }
+        ? symbol extends S? {} : { [P in S]: never } : {}
+}
```
Comment (jakebailey) [inline types/readable-stream/index.d.ts:14]:
```suggestion
type Is<T extends U, U> = T;
declare var NoAsyncDispose: {
    new(...arguments: any[]): typeof globalThis.Symbol extends { readonly iterator: Is<infer S, symbol> }
        ? symbol extends S ? {} : { [P in S]: never } : {}
}
```

File: types/readable-stream/package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
     "projects": [
         "https://github.com/nodejs/readable-stream"
     ],
+    "minimumTypeScriptVersion": "4.7",
```
Comment (jakebailey) [inline types/readable-stream/package.json:8]:
```suggestion
```

**Review by jakebailey — COMMENTED** [review]
Sorry it took so long to reply; I think these suggested changes should make it work back before 4.7.

File: types/readable-stream/index.d.ts (hunk 1 above)
Comment (Semigradsky) [reply types/readable-stream/index.d.ts]: Thank you! Looks good

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/readable-stream/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c3e6fbf37d73ef8852b09eb6bf409a34c2d787fa
</p>

**Review by jakebailey — APPROVED** [review]

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**